crates/proto/src/xfer/dnssec_dns_handle.rs

```diff
@@ -322,54 +320,41 @@ where
         // verify this rrset
         let proof = verify_rrset(handle.clone_with_context(), rrset, rrsigs, options).await;
 
-        let proof = match proof {
-            Ok(proof) => {
+        let (proof, adjusted_ttl) = match proof {
+            Ok((proof, adjusted_ttl)) => {
                 debug!("verified: {name} record_type: {record_type}",);
-                proof
+                (proof, adjusted_ttl)
             }
             Err(ProofError { proof, kind }) => {
                 debug!("failed to verify: {name} record_type: {record_type}: {kind}",);
-                proof
+                (proof, None)
             }
         };
 
-        rrset_proofs.insert((name, record_type), proof);
+        rrset_proofs.insert((name, record_type), (proof, adjusted_ttl));
     }
 
-    // assign TTL for all records that match name and record type.
-    let mut rrsig_ttls: HashMap<(Name, RecordType), u32> = HashMap::new();
-
     // set the proofs of all the records, all records are returned, it's up to downstream users to check for correctness
     let mut records = records;
     for record in &mut records {
         // the RRSIG used to validate a record inherits the outcome of the validation
         // for RRSIGs, we need to use their TYPE_COVERED field instead of `RecordType::RRSIG` as the
         // `RecordType` key in `rrset_proofs`
         let record_type = if let RData::DNSSEC(DNSSECRData::RRSIG(rrsig)) = record.data() {
-            let record_type = rrsig.type_covered();
```

Comment:
It makes it harder to review that this commit undoes a number of changes from the previous commit, would it be possible to restructure that? Maybe just get rid of the previous commit?

Reply:
Ok, commit 6e4b7e0d10baca853f82ed2a41878596c728f74f has been squashed into ac2eb08199c22a74917cc4eead975a8d77657db0.